core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -1349,6 +1434,25 @@ export class PolyfaceQuery {
     BuildAverageNormalsContext.buildFastAverageNormals(polyface, toleranceAngle);
   }
 
+  /**
+   * Offset the faces of the mesh.
+   * @param source original mesh
+   * @param signedOffsetDistance distance to offset
+   * @param offsetOptions angle options.  The default options are recommended.
+   * @returns shifted mesh.
+   */
+  public static cloneOffset(source: IndexedPolyface,
```

Comment:
Any reason why this and `convertToHalfEdgeGraph` are not methods of `IndexedPolyface` instead of static methods of `PolyfaceQuery`?

Reply:
I'm on the fence about that.  I've come to like to have complicated algorithms outside the object class.  Living in PolyfaceQuery fits with that.  (And midway through the life of imodel I've shifted form XxxxQuery to XxxxOps).